packages/blade/src/components/ActionList/ActionListFooter.tsx

```diff
@@ -0,0 +1,108 @@
+import React from 'react';
+import styled from 'styled-components';
+import { componentIds } from './componentIds';
+import Box from '~components/Box';
+import type { IconComponent } from '~components/Icons';
+import { useDropdown } from '~components/Dropdown/useDropdown';
+import { isReactNative, makeSize } from '~utils';
+import { BaseText } from '~components/Typography/BaseText';
+
+type ActionListFooterProps = {
+  title?: string;
+  leading?: React.ReactNode;
+  trailing?: React.ReactNode;
+};
+
+const StyledActionListFooter = styled(Box)((props) => {
+  return {
+    display: 'flex',
+    flexDirection: 'row',
+    alignItems: 'center',
+    padding: `${makeSize(props.theme.spacing[3])} ${makeSize(props.theme.spacing[5])}`,
+    backgroundColor: props.theme.colors.brand.gray.a50.lowContrast,
+    margin: `${makeSize(props.theme.spacing[3])} -${makeSize(props.theme.spacing[3])} -${makeSize(
+      props.theme.spacing[3],
+    )}`,
+  };
+});
+
+const ActionListFooter = (props: ActionListFooterProps): JSX.Element => {
+  const footerRef = React.useRef<HTMLDivElement | null>(null);
+  const {
+    setShouldIgnoreBlur,
+    setHasFooterAction,
+    onSelectKeydown,
+    activeIndex,
+    setIsOpen,
+  } = useDropdown();
+
+  React.useEffect(() => {
+    if (!isReactNative() && footerRef.current?.querySelector('button, a')) {
+      setHasFooterAction(true);
+    }
+  }, [setHasFooterAction, props.trailing]);
+
+  const isOnlyActionButton = !props.title && !props.leading && props.trailing;
+
+  return (
+    <StyledActionListFooter
+      // eslint-disable-next-line @typescript-eslint/no-explicit-any
+      ref={footerRef as any}
+      // eslint-disable-next-line
```

Comment:
Let's disable this eslint rule for the file. 

Plus it might just be better to define the type on the StyledActionListFooter itself, even if we are not using it on native. 

```ts
const StyledActionListFooter = styled(Box)<{ onMouseDown: ..., onKeyDown: ..., onBlur: ... }>((props) => {
```

Reply:
Disabled the ban-ts rule for this file. I tried adding types on footer but that too was failing with some big overload errors. Didn't spend much time on it.